### ISSUE TITLE
Same observation being generated

### ISSUE THREAD

**Issue by timothyjgraham** [yanyachen/rBayesianOptimization#35]
Hi @yanyachen,

I have encountered the same bug in the R implementation as in the python version: https://github.com/fmfn/BayesianOptimization/issues/10 .

When I optimise the XGBoost model hyper-parameters, the first observation is OK, but from the 2nd observation onwards I simply get the same value being returned over and over.

I have tried different optimisation functions, e.g., AUC, f1-score, accuracy, etc. Always the same problem.

I have also tried all the solutions from the linked Issue above, where possible.

Please advise and thanks for your time.
